core/src/autogluon/core/utils/utils.py

```diff
@@ -21,86 +21,89 @@
 from ..constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS, QUANTILE
 from ..metrics import accuracy, root_mean_squared_error, pinball_loss, Scorer
 
-logger = logging.getLogger(__name__)
-
 
+logger = logging.getLogger(__name__)
 
-# TODO: RespirceManager to replace these two hacks
-class _get_cpu_count:
-    def __init__(self):
-        self.mock = None
 
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
+class ResourceManager():
+    """Manager that fetches system related info"""
+    
+    @staticmethod
+    def get_cpu_count():
```

Comment:
In several areas of the code we actually use psutil to get num cpus, lets try moving these usages to `def get_cpu_count_psutil(logical=True)` in ResourceManager so it is easier to track.

Example in LightGBM:

```
num_cpus = psutil.cpu_count(logical=False)
```